src/WebJobs.Script/Binding/GeneralScriptBindingProvider.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Reflection;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Extensibility;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Script.Binding
+{
+    // This single binder can service all SDK extensions by leveraging the SDK metadata provider. 
+    public class GeneralScriptBindingProvider : ScriptBindingProvider
+    {        
+        private IJobHostMetadataProvider _tooling;
+                
+        public GeneralScriptBindingProvider(
+            JobHostConfiguration config, 
+            JObject hostMetadata, 
+            TraceWriter traceWriter) :
+            base(config, hostMetadata, traceWriter)
+        {
+        }
+
+        // The constructor is fixed and ScriptBindingProvider are instantated for us by the Script runtime. 
+        // Extensions may get registered after this class is instantiated. 
+        // So we need a final call that lets us get the tooling snapshot of the graph after all extensions are set. 
+        public void FinishInit()
+        {
+            this._tooling  = this.Config.GetTooling();
+        }
+
+        public override bool TryCreate(ScriptBindingContext context, out ScriptBinding binding)
+        {
+            string name = context.Type;
+            var attrType = this._tooling.GetAttributeTypeFromName(name);
+            if (attrType == null)
+            {
+                binding = null;
+                return false;
+            }
+
+            var attr = this._tooling.GetAttribute(attrType, context.Metadata);
+                        
+            binding = new GeneralScriptBinding(this._tooling, attr, context);
+            return true;
+        }
+
+        public override bool TryResolveAssembly(string assemblyName, out Assembly assembly)
+        {
+            return this._tooling.TryResolveAssembly(assemblyName, out assembly);
+        }
+
```

Comment:
Fix formatting here and throughout - bunch of extra spaces. Actually, if you rebase, you should see that Fabio's new analyzer stuff should generate build errors for these.